### PR TITLE
Add configuration class & tests

### PR DESCRIPTION
**Pull request by chevinbrown**
- Now defaults to using `:jbuilder` instead of serializers.
- Learned there was zero test coverage. Will need to write tests here also...

- [x] Add basic response tests
- [x] Fix up hound config
- [x] Fix up Readme

**Questions**
- Extra config options?
- Should the config class be an `attr_accessor`?
- How should we test `unauthenticated`?

### CODE REVIEW

File: test/json_api_responders_test.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    end
+
+    r = JSON.parse(response.body)
+    assert_equal 422, r['status']
+    assert_equal 'Missing Parameter', r['message']
+    assert_equal 'post', r['resource']
+    assert_equal 'Please supply the post param', r['detail']
+  end
+
+  # test 'rescue Unauthorized' do
```
Comment (dwilkins) [inline test/json_api_responders_test.rb]: :scissors:

File: lib/json_api_responders/responder.rb [hunk 2]
```diff
@@ -103,5 +111,20 @@ def resource_name
...
       I18n.t('json_api.errors.conflict.reason')
     end
+
+    def added_errors_obj(error_obj)
+      error_obj[:errors] = error_response unless errors
+    end
+
+    def added_resource(error_obj)
+      return unless errors
+      return error_obj[:resource] = resource.model if resource.try(:model).present?
```
Comment (dwilkins) [inline lib/json_api_responders/responder.rb]: Need safe accessor &. for `resource` since it can be  nil from `respond_with_error`

**Review by dwilkins — CHANGES_REQUESTED** [review]
Just one small change - :+1: for adding tests and infrastructure associated.   

A more thorough review of the existing code may be warranted.

File: json_api_responders.gemspec [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
-$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+$:.push File.expand_path("../lib", __FILE__)
 require 'json_api_responders/version'
 
 Gem::Specification.new do |spec|
   spec.name = 'json_api_responders'
   spec.version = JsonApiResponders::VERSION
-  spec.authors = ['Stanko Krtalić Rusendić']
-  spec.email = ['stanko.krtalic@gmail.com']
+  spec.authors = ['Kevin Brown']
```
Comment (licatajustin) [inline json_api_responders.gemspec]: Please keep the original author in here as well.

File: lib/json_api_responders/responder.rb (hunk 2 above)
Comment (licatajustin) [reply lib/json_api_responders/responder.rb]: (to comment above regarding `dig`) Be careful with that operator, as it doesn't has 2.1 support. This gem will support 2.1+.

File: README.md [hunk 4]
```diff
@@ -28,6 +28,14 @@ end
 
 ## Usage
 
+Configure with an initializer to pick the default responder_type `:jbuilder` or `:serializer`. `:jbuilder` is the default.
```
Comment (licatajustin) [inline README.md]: This README needs detailed description of JSON format.

**Review by licatajustin — CHANGES_REQUESTED** [review]

**Review by dwilkins — APPROVED** [review]

**Pull request closed without merging**